src/mpyl/run_plan.py

```diff
@@ -182,29 +206,36 @@ def load_from_pickle_file():
                 f"Unable to find existing run plan at path {RUN_PLAN_PICKLE_FILE}"
             )
 
-    def print_markdown(self, console: Console, stages: list[Stage]):
-        if self.has_projects_to_run(include_cached_projects=True):
+    def to_markdown(self, execution_result: Optional[ExecutionResult] = None) -> str:
+        if self._has_projects_to_run(include_cached_projects=True):
             result = ""
 
-            for stage in stages:
-                executions = self.get_executions_for_stage(stage)
+            for stage in self._get_all_stages():
+                result += f"{stage.to_markdown()}:  \n"
```

Comment:
Another great catch! 💪 

To be honest: I don't know. The version currently in `main` is very broken:
<img width="546" alt="image" src="https://github.com/user-attachments/assets/65b46417-e797-4ec1-bc5c-674fa4d1a8ca">

This change makes it so so that it at least prints which stage it refers to:
<img width="563" alt="image" src="https://github.com/user-attachments/assets/5552b681-0012-4150-9ee1-d97372cb712d">

This at least makes it more clear that _"the test stage is part of the run plan (via another project), but the project we selected has nothing to do in that stage."_

We can also do:
1. omit stages where the selected project is not meant to run (in the example above, it would hide _Test_ and _Postdeploy_)
2. stop printing the entire run plan when calling `mpyl build run --stage deploy --project deploy`, and print only the selected stage and project
 
What do you prefer?

Reply:
Ah, wait, the plot thickens!

So it turns out I was partially wrong: the two screenshots above are about **_discovering_** a new run plan, and will of course show every project in each configured stage. In this case it's probably a good idea to hide stages without any project _(what's the point of printing a stage without any project in it?)_.

Later on, when we're actually running a project in a stage, **_then_** my previous comment applies (and the question still holds).

I'll add some more unit tests to illustrate this 👍

Reply:
Actually I went in the opposite direction: instead of making the code extremely complicated to align with the current outputs (which most people don't even read), I decided to make those outputs simpler and clearer:
* [if the run succeeds](https://github.com/Vandebron/gh-mpyl/pull/105/files#diff-db3a4144e2c7927739b2d56958d580cd13aa1b4c1744df6806bb7f735dd380c0)
* [if the run fails](https://github.com/Vandebron/gh-mpyl/pull/105/files#diff-a1c0d2e5a68b333b58c0101a1d7c47b78be932319f876fd17135f6750f4c4e20)
* [if the run fails with an exception](https://github.com/Vandebron/gh-mpyl/pull/105/files#diff-15b9ad4d1586e34b641b9c948ca037ff8a887214d9ac697fc51aa8b225aa56de)

This makes the code simpler and reduces the size of this PR 💪